The process '/usr/bin/gpg' failed with exit code 2

**Describe the bug**

`setup-swift` fails on `ubuntu-latest` with `The process '/usr/bin/gpg' failed with exit code 2`. This looks like #77 reappeared. Full logs can be found below.

**Workflow configuration (please complete the following information):**
 - Platform (`runs-on`): `ubuntu-latest`
 - Swift version (`swift-version`): any, see 5.0 as example in logs below.

```
Run fwal/setup-swift@v1.11.0
  with:
    swift-version: 5.0
/usr/bin/gpg --import /home/runner/work/_temp/6f3cd87a-c90d-48ae-a6a4-9509bc4160d0
gpg: directory '/home/runner/.gnupg' created
gpg: keybox '/home/runner/.gnupg/pubring.kbx' created
gpg: /home/runner/.gnupg/trustdb.gpg: trustdb created
gpg: key D441C977412B37AD: public key "Swift Automatic Signing Key #1 <swift-infrastructure@swift.org>" imported
gpg: key 9F597F4D21A56D5F: public key "Swift 2.2 Release Signing Key <swift-infrastructure@swift.org>" imported
gpg: key 63BC1CFE91D306C6: public key "Swift 3.x Release Signing Key <swift-infrastructure@swift.org>" imported
gpg: key EF5430F071E1B235: public key "Swift 4.x Release Signing Key <swift-infrastructure@swift.org>" imported
gpg: key 7638F1FB2B2B08C4: public key "Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>" imported
gpg: key 925CC1CCED3D1561: public key "Swift 5.x Release Signing Key <swift-infrastructure@swift.org>" imported
gpg: key FAF6989E1BC16FEA: public key "Swift Automatic Signing Key #3 <swift-infrastructure@swift.org>" imported
gpg: key 925CC1CCED3D1561: "Swift 5.x Release Signing Key <swift-infrastructure@swift.org>" 1 new signature
gpg: key F167DF1ACF9CE069: public key "Swift Automatic Signing Key #4 <swift-infrastructure@forums.swift.org>" imported
gpg: Total number processed: 9
gpg:               imported: 8
gpg:         new signatures: 1
/usr/bin/gpg --keyserver hkp://pool.sks-keyservers.net --refresh-keys Swift
gpg: refreshing 8 keys from hkp://pool.sks-keyservers.net
gpg: keyserver refresh failed: No name
Warning: An error occurred when trying to refresh keys from hkp://pool.sks-keyservers.net: Error: The process '/usr/bin/gpg' failed with exit code 2
/usr/bin/gpg --keyserver ha.pool.sks-keyservers.net --refresh-keys Swift
gpg: refreshing 8 keys from hkp://ha.pool.sks-keyservers.net
gpg: keyserver refresh failed: No name
Warning: An error occurred when trying to refresh keys from ha.pool.sks-keyservers.net: Error: The process '/usr/bin/gpg' failed with exit code 2
/usr/bin/gpg --keyserver keyserver.ubuntu.com --refresh-keys Swift
gpg: refreshing 8 keys from hkp://keyserver.ubuntu.com
gpg: key F167DF1ACF9CE069: "Swift Automatic Signing Key #4 <swift-infrastructure@forums.swift.org>" not changed
gpg: key FAF6989E1BC16FEA: "Swift Automatic Signing Key #3 <swift-infrastructure@swift.org>" not changed
gpg: key 925CC1CCED3D1561: "Swift 5.x Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key 7638F1FB2B2B08C4: "Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>" not changed
gpg: key EF5430F071E1B235: "Swift 4.x Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key 63BC1CFE91D306C6: "Swift 3.x Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key 9F597F4D21A56D5F: "Swift 2.2 Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key D441C977412B37AD: "Swift Automatic Signing Key #1 <swift-infrastructure@swift.org>" not changed
gpg: Total number processed: 8
gpg:              unchanged: 8
/usr/bin/gpg --verify /home/runner/work/_temp/e10b5e90-937d-4a19-a17b-03c60e7bbdd5 /home/runner/work/_temp/7a6475e2-2771-4bf1-a368-ed6196ce9cd6
gpg: no valid OpenPGP data found.
gpg: the signature could not be verified.
Please remember that the signature file (.sig or .asc)
should be the first file given on the command line.
Error: Unexpected error, unable to continue. Please report at https://github.com/fwal/setup-swift/issues
The process '/usr/bin/gpg' failed with exit code 2
Stacktrace:
Error: The process '/usr/bin/gpg' failed with exit code 2
    at ExecState._setResult (/home/runner/work/_actions/fwal/setup-swift/v1.11.0/dist/index.js:995:25)
    at ExecState.CheckComplete (/home/runner/work/_actions/fwal/setup-swift/v1.11.0/dist/index.js:978:18)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/fwal/setup-swift/v1.11.0/dist/index.js:872:27)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Socket.<anonymous> (internal/child_process.js:430:11)
    at Socket.emit (events.js:210:5)
    at Pipe.<anonymous> (net.js:659:12)
```

I'm also encountering this issue, and I can confirm that it is the same issue as #77. And if you look at the log excerpt that I've included below, you can see that despite setup-swift consulting a pool of key servers (as was discussed in the resolution to the original issue), it fails on most of them:
```
/usr/bin/gpg --keyserver hkp://pool.sks-keyservers.net --refresh-keys Swift
gpg: refreshing 8 keys from hkp://pool.sks-keyservers.net
gpg: keyserver refresh failed: Server indicated a failure
Warning: An error occurred when trying to refresh keys from hkp://pool.sks-keyservers.net: Error: The process '/usr/bin/gpg' failed with exit code 2
```
```
/usr/bin/gpg --keyserver ha.pool.sks-keyservers.net --refresh-keys Swift
gpg: refreshing 8 keys from hkp://ha.pool.sks-keyservers.net
gpg: keyserver refresh failed: Server indicated a failure
Warning: An error occurred when trying to refresh keys from ha.pool.sks-keyservers.net: Error: The process '/usr/bin/gpg' failed with exit code 2
```
and on the one server that works, none of the keys are updated:
```
/usr/bin/gpg --keyserver keyserver.ubuntu.com --refresh-keys Swift
gpg: refreshing 8 keys from hkp://keyserver.ubuntu.com
gpg: key F167DF1ACF9CE069: "Swift Automatic Signing Key #4 <swift-infrastructure@forums.swift.org>" not changed
gpg: key FAF6989E1BC16FEA: "Swift Automatic Signing Key #3 <swift-infrastructure@swift.org>" not changed
gpg: key 925CC1CCED3D1561: "Swift 5.x Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key 7638F1FB2B2B08C4: "Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>" not changed
gpg: key EF5430F071E1B2[35](https://github.com/Samasaur1/SwiftIP/runs/5088383035?check_suite_focus=true#step:3:35): "Swift 4.x Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key 63BC1CFE91D306C6: "Swift 3.x Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key 9F597F4D21A56D5F: "Swift 2.2 Release Signing Key <swift-infrastructure@swift.org>" not changed
gpg: key D441C977412B37AD: "Swift Automatic Signing Key #1 <swift-infrastructure@swift.org>" not changed
gpg: Total number processed: 8
gpg:              unchanged: 8
```

<details>
  <summary>Complete log</summary>
  
```
2022-02-07T04:53:56.7541964Z ##[group]Run fwal/setup-swift@v1
2022-02-07T04:53:56.7542189Z with:
2022-02-07T04:53:56.7542384Z   swift-version: 5.1
2022-02-07T04:53:56.7542584Z ##[endgroup]
2022-02-07T04:53:57.2030543Z [command]/usr/bin/gpg --import /home/runner/work/_temp/41047049-be27-47b4-88d1-aeddc7ec71d7
2022-02-07T04:53:57.2098387Z gpg: directory '/home/runner/.gnupg' created
2022-02-07T04:53:57.2099448Z gpg: keybox '/home/runner/.gnupg/pubring.kbx' created
2022-02-07T04:53:57.2118386Z gpg: /home/runner/.gnupg/trustdb.gpg: trustdb created
2022-02-07T04:53:57.2119663Z gpg: key D441C977412B37AD: public key "Swift Automatic Signing Key #1 <swift-infrastructure@swift.org>" imported
2022-02-07T04:53:57.2173851Z gpg: key 9F597F4D21A56D5F: public key "Swift 2.2 Release Signing Key <swift-infrastructure@swift.org>" imported
2022-02-07T04:53:57.2185229Z gpg: key 63BC1CFE91D306C6: public key "Swift 3.x Release Signing Key <swift-infrastructure@swift.org>" imported
2022-02-07T04:53:57.2197650Z gpg: key EF5430F071E1B235: public key "Swift 4.x Release Signing Key <swift-infrastructure@swift.org>" imported
2022-02-07T04:53:57.2204686Z gpg: key 7638F1FB2B2B08C4: public key "Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>" imported
2022-02-07T04:53:57.2211797Z gpg: key 925CC1CCED3D1561: public key "Swift 5.x Release Signing Key <swift-infrastructure@swift.org>" imported
2022-02-07T04:53:57.2218666Z gpg: key FAF6989E1BC16FEA: public key "Swift Automatic Signing Key #3 <swift-infrastructure@swift.org>" imported
2022-02-07T04:53:57.2317716Z gpg: key 925CC1CCED3D1561: "Swift 5.x Release Signing Key <swift-infrastructure@swift.org>" 1 new signature
2022-02-07T04:53:57.2328671Z gpg: key F167DF1ACF9CE069: public key "Swift Automatic Signing Key #4 <swift-infrastructure@forums.swift.org>" imported
2022-02-07T04:53:57.2333695Z gpg: Total number processed: 9
2022-02-07T04:53:57.2334179Z gpg:               imported: 8
2022-02-07T04:53:57.2437461Z gpg:         new signatures: 1
2022-02-07T04:53:57.2438576Z [command]/usr/bin/gpg --keyserver hkp://pool.sks-keyservers.net --refresh-keys Swift
2022-02-07T04:53:57.4917181Z gpg: refreshing 8 keys from hkp://pool.sks-keyservers.net
2022-02-07T04:55:01.8269699Z gpg: keyserver refresh failed: Server indicated a failure
2022-02-07T04:55:01.8308710Z ##[warning]An error occurred when trying to refresh keys from hkp://pool.sks-keyservers.net: Error: The process '/usr/bin/gpg' failed with exit code 2
2022-02-07T04:55:01.8319468Z [command]/usr/bin/gpg --keyserver ha.pool.sks-keyservers.net --refresh-keys Swift
2022-02-07T04:55:01.8349264Z gpg: refreshing 8 keys from hkp://ha.pool.sks-keyservers.net
2022-02-07T04:56:21.9221359Z gpg: keyserver refresh failed: Server indicated a failure
2022-02-07T04:56:21.9233500Z ##[warning]An error occurred when trying to refresh keys from ha.pool.sks-keyservers.net: Error: The process '/usr/bin/gpg' failed with exit code 2
2022-02-07T04:56:21.9242917Z [command]/usr/bin/gpg --keyserver keyserver.ubuntu.com --refresh-keys Swift
2022-02-07T04:56:21.9283886Z gpg: refreshing 8 keys from hkp://keyserver.ubuntu.com
2022-02-07T04:56:25.6621123Z gpg: key F167DF1ACF9CE069: "Swift Automatic Signing Key #4 <swift-infrastructure@forums.swift.org>" not changed
2022-02-07T04:56:25.6625624Z gpg: key FAF6989E1BC16FEA: "Swift Automatic Signing Key #3 <swift-infrastructure@swift.org>" not changed
2022-02-07T04:56:25.6636696Z gpg: key 925CC1CCED3D1561: "Swift 5.x Release Signing Key <swift-infrastructure@swift.org>" not changed
2022-02-07T04:56:25.6643144Z gpg: key 7638F1FB2B2B08C4: "Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>" not changed
2022-02-07T04:56:25.6649068Z gpg: key EF5430F071E1B235: "Swift 4.x Release Signing Key <swift-infrastructure@swift.org>" not changed
2022-02-07T04:56:25.6655726Z gpg: key 63BC1CFE91D306C6: "Swift 3.x Release Signing Key <swift-infrastructure@swift.org>" not changed
2022-02-07T04:56:25.6661743Z gpg: key 9F597F4D21A56D5F: "Swift 2.2 Release Signing Key <swift-infrastructure@swift.org>" not changed
2022-02-07T04:56:25.6675250Z gpg: key D441C977412B37AD: "Swift Automatic Signing Key #1 <swift-infrastructure@swift.org>" not changed
2022-02-07T04:56:25.6675619Z gpg: Total number processed: 8
2022-02-07T04:56:25.6675838Z gpg:              unchanged: 8
2022-02-07T04:56:26.1590532Z [command]/usr/bin/gpg --verify /home/runner/work/_temp/0dd3ba39-d930-456a-a78f-df98dd32a0be /home/runner/work/_temp/94ed0956-c297-403d-882d-d8ae733374e0
2022-02-07T04:56:26.1630085Z gpg: no valid OpenPGP data found.
2022-02-07T04:56:26.1631098Z gpg: the signature could not be verified.
2022-02-07T04:56:26.1631661Z Please remember that the signature file (.sig or .asc)
2022-02-07T04:56:26.1632696Z should be the first file given on the command line.
2022-02-07T04:56:26.1652446Z ##[error]Unexpected error, unable to continue. Please report at https://github.com/fwal/setup-swift/issues
The process '/usr/bin/gpg' failed with exit code 2
Stacktrace:
Error: The process '/usr/bin/gpg' failed with exit code 2
    at ExecState._setResult (/home/runner/work/_actions/fwal/setup-swift/v1/dist/index.js:995:25)
    at ExecState.CheckComplete (/home/runner/work/_actions/fwal/setup-swift/v1/dist/index.js:978:18)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/fwal/setup-swift/v1/dist/index.js:872:27)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```
(from https://github.com/Samasaur1/SwiftIP/runs/5088383035)
</details>

--- a/__tests__/gpg.test.ts
+++ b/__tests__/gpg.test.ts
@@ -16,13 +16,31 @@ describe("gpg", () => {
     expect(mockExec).toBeCalledTimes(1);
   });
 
-  it("uses the next keyserver in the pool if the previous fails", async () => {
-    const failingServers = 3;
-    let testedServers = 0;
+  // NOTE: Currently disabled as the pool only contains one server
+  // it("uses the next keyserver in the pool if the previous fails", async () => {
+  //   const failingServers = 3;
+  //   let testedServers = 0;
+
+  //   mockExec.mockImplementation(() => {
+  //     testedServers++;
+  //     if (testedServers >= failingServers) {
+  //       return Promise.resolve(0);
+  //     } else {
+  //       return Promise.resolve(1);
+  //     }
+  //   });
+
+  //   await refreshKeys();
+  //   expect(mockExec).toBeCalledTimes(3);
+  // });
+
+  it("makes a second attempt if the keyserver fails", async () => {
+    const attempts = 2;
+    let tests = 0;
 
     mockExec.mockImplementation(() => {
-      testedServers++;
-      if (testedServers >= failingServers) {
+      tests++;
+      if (tests >= attempts) {
         return Promise.resolve(0);
       } else {
         return Promise.resolve(1);
@@ -30,7 +48,7 @@ describe("gpg", () => {
     });
 
     await refreshKeys();
-    expect(mockExec).toBeCalledTimes(3);
+    expect(mockExec).toBeCalledTimes(2);
   });
 
   it("throws an error if all servers in the pool fails", async () => {

--- a/src/gpg.ts
+++ b/src/gpg.ts
@@ -25,8 +25,15 @@ export async function refreshKeys() {
 
   for (const server of pool) {
     core.debug(`Refreshing keys from ${server}`);
+    // 1st try...
     if (await refreshKeysFromServer(server)) {
-      core.debug(`Refresh successful`);
+      core.debug(`Refresh successful on first attempt`);
+      return;
+    }
+
+    // 2nd try...
+    if (await refreshKeysFromServer(server)) {
+      core.debug(`Refresh successful on second attempt`);
       return;
     }
     core.debug(`Refresh failed`);

--- a/src/gpg.ts
+++ b/src/gpg.ts
@@ -21,13 +21,7 @@ export async function verify(signaturePath: string, packagePath: string) {
 }
 
 export async function refreshKeys() {
-  const pool = [
-    "hkp://pool.sks-keyservers.net",
-    "ha.pool.sks-keyservers.net",
-    "keyserver.ubuntu.com",
-    "hkp://keyserver.ubuntu.com",
-    "pgp.mit.edu",
-  ];
+  const pool = ["hkp://keyserver.ubuntu.com"];
 
   for (const server of pool) {
     core.debug(`Refreshing keys from ${server}`);

